Install failed: EACCES: permission denied

https://circleci.com/gh/Leko/hothouse/258

```
#!/bin/bash -eo pipefail
npm install -g hothouse
/usr/local/bin/hothouse -> /usr/local/lib/node_modules/hothouse/dist/cli.js

> parse-domain@2.1.2 postinstall /usr/local/lib/node_modules/hothouse/node_modules/parse-domain
> node scripts/build-tries.js

Downloading public suffix list from https://publicsuffix.org/list/public_suffix_list.dat... ok
Writing /usr/local/lib/node_modules/hothouse/node_modules/parse-domain/build/tries/current/icann.complete.json... 
Could not update list of known top-level domains for parse-domain because of EACCES: permission denied, mkdir '/usr/local/lib/node_modules/hothouse/node_modules/parse-domain/build/tries/current'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! parse-domain@2.1.2 postinstall: `node scripts/build-tries.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the parse-domain@2.1.2 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-07-05T15_11_03_784Z-debug.log
Exited with code 1
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -37,9 +37,13 @@ jobs:
       - run: npm run flow
       - run: npm test
       - run: |
-          mkdir /tmp/hothouse-install-test
-          cd /tmp/hothouse-install-test
+          mkdir /tmp/hothouse-install-locally
+          cd /tmp/hothouse-install-locally
           npm install $CIRCLE_WORKING_DIRECTORY/packages/hothouse
+      - run: |
+          mkdir /tmp/hothouse-install-globally
+          cd /tmp/hothouse-install-globally
+          npm install -g $CIRCLE_WORKING_DIRECTORY/packages/hothouse
       - run: bash <(curl -s https://codecov.io/bash)
   node-v8:
     <<: *base

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -36,6 +36,10 @@ jobs:
       - run: npm run lint
       - run: npm run flow
       - run: npm test
+      - run: |
+          mkdir /tmp/hothouse-install-test
+          cd /tmp/hothouse-install-test
+          npm install $CIRCLE_WORKING_DIRECTORY/packages/hothouse
       - run: bash <(curl -s https://codecov.io/bash)
   node-v8:
     <<: *base

--- a/packages/hothouse/package.json
+++ b/packages/hothouse/package.json
@@ -38,7 +38,6 @@
     "@octokit/rest": "^15.9.2",
     "chalk": "^2.4.1",
     "debug": "^3.1.0",
-    "git-url-parse": "^9.0.1",
     "glob": "^7.1.2",
     "hosted-git-info": "^2.6.1",
     "isomorphic-git": "^0.19.11",

--- a/packages/hothouse/src/Hosting/GitHub.js
+++ b/packages/hothouse/src/Hosting/GitHub.js
@@ -1,12 +1,19 @@
 // @flow
-import parse from "git-url-parse";
+import { URL } from "url";
 import octokit from "@octokit/rest";
 import type { Hosting } from "@hothouse/types";
+import { fromUrl } from "hosted-git-info";
 import { getTagsQuery } from "./graphql";
 
 export const parseRepositoryUrl = (url: string): [string, string] => {
-  const { owner, name } = parse(url);
-  return [owner, name];
+  const parsed = fromUrl(url);
+  if (!parsed) {
+    const urlObj = new URL(url);
+    const [owner, repo] = urlObj.pathname.split("/").slice(1);
+    return [owner, repo];
+  }
+  const { user, project } = parsed;
+  return [user, project];
 };
 
 export default class GitHub implements Hosting {

